graylog2-server/src/main/java/org/graylog2/streams/StreamRouterEngine.java

```diff
@@ -258,8 +260,11 @@ public void increment() {
         }
 
         public boolean isMatched() {
-            // If a stream has multiple stream rules, all of the rules have to match.
-            return ruleCount == matches;
+            switch (stream.getMatchingType()) {
```

Comment:
How about storing the matching type in a field in the constructor? That would avoid calling `stream.getMatchingType()` over and over again.